modules/codegen/src/smithy4s/codegen/SmithyToIR.scala

```diff
@@ -480,9 +525,21 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
   private object UnRef {
     def unapply(tpe: Type): Option[Shape] = tpe match {
       case Type.Ref(ns, name) =>
-        model
+        val maybeShape = model
           .getShape(ShapeId.fromParts(ns, name))
           .asScala
+        maybeShape.map { shape =>
+          shape.getTrait(classOf[AdtMemberTrait]).asScala match {
+            case Some(adtMemberTrait) =>
+              val cId = shape.getId
+              val newNs =
+                cId.getNamespace + "." + adtMemberTrait.getValue.getName
+              shape.asStructureShape.get.toBuilder
```

Comment:
I see a `.get` I get goosebumps ;)

jokes aside, I think it would be an improvement to use `map / getOrElse { throw RelevantError }`

it would be equivalent (throwing on none), but at least the error message would make some sense

Reply:
If a shape has the AdtMemberTrait, it has to be a structure shape.. I could add logic to throw an error like you say, but it feels unnecessary when that error will never be thrown 🤷

Reply:
we're saying that the error will never be thrown (because honestly it most likely never will in the context the *trait exists* to support structure special case), but there are always chances that this happen and an error like "`AdtMemberTrait` should be applied to structure only" is better than `NoSuchElementException`

it's very minor, and to be honest, I don't mind much so I'll trust your judgement, but I've search through a fair share of `no such element` exceptions that I never do that anymore

Reply:
Yeah, I mean doesn't hurt to add other than being a bit more verbose, but not enough for me to worry. I'll update so it has a more specific error message